packages/js/plugins/ethereum/src/index.ts

```diff
@@ -67,10 +67,42 @@ export class EthereumPlugin extends Module<EthereumPluginConfig> {
     _client: Client
   ): Promise<string> {
     const connection = await this._getConnection(args.connection);
-    const contract = connection.getContract(args.address, [args.method], false);
+    let method, isRawAbiMethod;
+    try {
+      method = JSON.parse(args.method);
+      isRawAbiMethod = true;
+    } catch (e) {
+      method = [args.method];
+      isRawAbiMethod = false;
+    }
+    const contract = connection.getContract(args.address, method, false);
     const funcs = Object.keys(contract.interface.functions);
     const res = await contract[funcs[0]](...parseArgs(args.args));
-    return res.toString();
+    if (!isRawAbiMethod) {
```

Comment:
Instead of having this logic here, I would suggest moving it within a `parseResult` function that can be reused. If we have this parsing problem for view methods, we most likely have the same problem for transaction methods as well.

So, in addition to moving to reusable function, we should use it in all other places where we're naively doing `.toString` on the result of a contract method call.

Reply:
agreed. The other two methods where we could use these, afaik, would be on `callContractStatic` and `_callContractMethod`. The last one is a proper tx, so we should be getting the results from `tx.data`.

Will add

Reply:
Probably also on `callContractMethodAndWait`, but `TxReceipt` doesnt have an attribute `data` to set the results :/

Reply:
hm, i assumed that i could get the return values from a transaction, but no i cant. Had never done that before. Only static calls return the values when the function is not `view`, therefore... seems like `callContractStatic` and `callContractView` would be the only two methods to parse results i think